eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandra.scala

```diff
@@ -52,60 +56,104 @@ object ReplicatedCassandra {
     statements: Statements[F],
     expiryService: ExpiryService[F],
   ): ReplicatedJournal[F] = {
+    applyWithTempMetrics(segmentSizeDefault, segmentNrsOf, statements, expiryService, None, None)
+  }
+
+  private def applyWithTempMetrics[F[_]: Sync: Parallel: Fail](
+    segmentSizeDefault: SegmentSize,
+    segmentNrsOf: SegmentNrsOf[F],
+    statements: Statements[F],
+    expiryService: ExpiryService[F],
+    metrics: Option[ReplicatedJournal.Metrics[F]],
+    log: Option[Log[F]],
+  ): ReplicatedJournal[F] = {
 
     new Main with ReplicatedJournal[F] {
 
-      def topics = {
+      val unit: F[Unit] = Applicative[F].unit
+
+      def topics: F[SortedSet[Topic]] = {
         statements
           .selectTopics2()
           .parProduct(statements.selectTopics())
-          .map { case (a, b) => a ++ b }
+          .flatMap {
+            case (a, b) =>
+              val meterAndLog = Applicative[F].whenA(b.diff(a).nonEmpty) {
```

Comment:
Minor thing:
Am I right that executing `meterAndLog` with `metrics == None` and `log == None` pointless?

Might be you can do something like:
```
metrics.product(log).traverse { case (metrics, log) =>
  ???
}
```

Reply:
Here's example what I'm talking about:
```
              val diff = b.diff(a)

              val actions = for {
                metrics <- metrics.toSeq
                log     <- log.toSeq
                topic   <- diff.toSeq
              } yield for {
                _ <- metrics.topicsFallback(topic)
                _ <- log.warn(s"`pointer` contains topic $topic while `pointer2` doesn't")
              } yield {}

              actions.sequence_.as(a ++ b)
```